services/venice-router/src/main/java/com/linkedin/venice/router/api/VeniceDelegateMode.java

```diff
@@ -221,20 +217,14 @@ public <H, P extends ResourcePath<K>, K, R> Scatter<H, P, K> scatter(
         RouterExceptionAndTrackingUtils
             .recordUnavailableReplicaStreamingRequest(storeName, venicePath.getRequestType());
       } else {
-        Collection<ScatterGatherRequest<H, K>> offlineRequests = scatter.getOfflineRequests();
-        StringBuilder partitions = new StringBuilder();
-
-        for (ScatterGatherRequest scatterGatherRequest: offlineRequests) {
-          partitions.append(scatterGatherRequest.getPartitionsNames());
-        }
         RouterExceptionAndTrackingUtils.FailureType failureType = RouterExceptionAndTrackingUtils.FailureType.REGULAR;
         if (venicePath.isRetryRequest()) {
           // don't record it as unhealthy request.
           failureType = RouterExceptionAndTrackingUtils.FailureType.RETRY_ABORTED_BY_NO_AVAILABLE_REPLICA;
         }
         String isRetry = venicePath.isRetryRequest() ? "retry " : "";
-        String errMsg = "Partitions : " + partitions + " not available to serve " + isRetry + "request of type: "
-            + venicePath.getRequestType();
+        String errMsg =
+            resourceName + " not available to serve " + isRetry + "request of type: " + venicePath.getRequestType();
```

Comment:
Can we keep this piece of logic for debugging purpose?
The non-available case won't happen at steady state and if it happens, the partition info in the log will help investigate the problem.

Reply:
I didn't remove this because of the overhead in the error-handling path, but because the collection returned by `scatterGatherRequest.getPartitionsNames()` appear to be populated only within `BROADCAST_BY_PARTITION` and `BROADCAST_BY_PRIMARY_HOST`, which I believe are unused strategies. Actually, I'd like to delete these two strategies entirely, and the partition names from `ScatterGatherRequest`, but there were a lot of test changes needed (I could still do it though). Essentially, the current code ends up instantiating a new `HashSet` for each `ScatterGatherRequest`, even though it is never used. It's possible that I didn't understand the code properly though. I am not sure.

Reply:
I think it is populated by the caller:
```
scatter.addOfflineRequest(new ScatterGatherRequest<>(Collections.emptyList(), keySet, partitionName));
```
If we want to get rid of `partitionNames` set from `ScatterGatherRequest`, I guess we can calculate the partition from one key of the offline `ScatterGatherRequest` on the fly, which should be lightweight.

Reply:
Oh yeah, that's right, that's where it was happening, and I removed it in this PR. Good idea to extract the partition from the first key... I'll do that.